SpatialGDK/Source/SpatialGDKFunctionalTests/SpatialGDK/SpatialTestRepNotify/SpatialTestRepNotify.cpp

```diff
@@ -24,47 +30,110 @@
 ASpatialTestRepNotify::ASpatialTestRepNotify()
 	: Super()
 {
-	Author = "Miron + Andrei";
+	Author = "Miron + Andrei + Arthur";
 	Description = TEXT("Test RepNotify replication and shadow data");
 }
 
 void ASpatialTestRepNotify::PrepareTest()
 {
 	Super::PrepareTest();
 
-	// The Server sets some initial values to the replicated variables.
-	AddStep(TEXT("SpatialTestRepNotifyServerSetReplicatedVariables"), FWorkerDefinition::Server(1), nullptr, [this]() {
-		OnChangedRepNotifyInt1 = 1;
-		AlwaysRepNotifyInt1 = 2;
-		OnChangedRepNotifyInt2 = 3;
-		AlwaysRepNotifyInt2 = 4;
-		TestArray.Empty();
-		TestArray.Add(1);
-		TestArray.Add(2);
+	/**
+	 * Here we check core rep notify functionality and the order rep notifies are called in.
+	 * In native, data will be applied for an actor, its subobjects, and only THEN will repnotifies be called for the updated data.
+	 *
+	 * To test this, we the update a replicated property on our actor and a subobject of that actor.
```

Comment:
not "we, the people", but "we the update" now

Reply:
wtf i literally cannot english

Reply:
changed